### PR TITLE
chore: Update `datafusion` / `arrow` / `parquet` to `45.0.0`

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* keep up with dependencies
* Closes https://github.com/influxdata/idpe/issues/17940
* Closes https://github.com/influxdata/idpe/issues/17990

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/27c7ae8978205dfebe7a96d6c1e28779df670bc2

Note that despite the small diff, the underlying libraries have some substantial changes to timestamp arithmetic (uses durations rather than intervals)

### CODE REVIEW

File: iox_query_influxql/src/window/difference.rs [hunk 1]
```diff
@@ -42,7 +43,7 @@ impl PartitionEvaluator for DifferencePartitionEvaluator {
         let array = Arc::clone(&values[0]);
         if array.null_count() == 0 {
             // If there are no gaps then use arrow kernels.
-            Ok(subtract_dyn(&array, &shift(&array, 1)?)?)
```
Comment (alamb) [inline iox_query_influxql/src/window/difference.rs:45]: subtract_dyn is deprecated in favor of sub_wrapping

**Review by appletreeisyellow — APPROVED** [review]
🚀

**Review by crepererum — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @appletreeisyellow  and @crepererum  -- I waited for this morning to roll this out during the work day. Slack thread is here: https://influxdata.slack.com/archives/CRK9M8L5Q/p1691586741920059

**Comment by alamb** [comment]
revert PR is prepared in case: https://github.com/influxdata/influxdb_iox/pull/8459

**Pull request merged**